packages/devtools_app/lib/src/network/network_controller.dart

```diff
@@ -100,28 +101,34 @@ class NetworkController with SearchControllerMixin<NetworkRequest> {
           !outstandingRequestsMap.containsKey(id)) {
         continue;
       }
-      httpEventIds.add(id);
-    }
 
-    // Group all HTTP timeline events with the same ID.
-    final httpEvents = <String, List<Map<String, dynamic>>>{};
-    for (final event in events) {
-      final json = event.toJson();
-      final id = json['id'];
-      if (id == null) {
-        continue;
-      }
-      if (httpEventIds.contains(id)) {
-        httpEvents.putIfAbsent(id, () => []).add(json);
+      // Any HTTP event with a specified 'parentId' is the response event of
+      // another request (the request with 'id' = 'parentId'). Store the
+      // relationship in [httpRequestIdToResponseId].
+      final parentId = json['args']['parentId'];
+      if (parentId != null) {
+        httpRequestIdToResponseId[parentId] = id;
       }
+      httpEvents.putIfAbsent(id, () => []).add(json);
     }
 
     // Build our list of network requests from the collected events.
     for (final request in httpEvents.entries) {
       final requestId = request.key;
+
+      // Do not handle response events - they are handled as part of the request
+      if (httpRequestIdToResponseId.values.contains(requestId)) continue;
+
+      final responseId = httpRequestIdToResponseId[requestId];
+      final responseEvents = <Map<String, dynamic>>[];
```

Comment:
prefer Map<String, Object> and add the required casts. that will catch more bugs relative to using dynamic.

Reply:
changed to object. What required casts were you referring to? I didn't see any errors when just changing from dynamic to Object